environment/src/main/java/com/sequenceiq/environment/credential/v1/converter/gcp/GcpCredentialV1ParametersToGcpCredentialAttributesConverter.java

```diff
@@ -38,10 +41,21 @@ private P12Attributes getP12(P12Parameters p12) {
 
     private JsonAttributes getJson(JsonParameters json) {
         JsonAttributes response = new JsonAttributes();
-        response.setCredentialJson(json.getCredentialJson());
+        String credentialJson = json.getCredentialJson();
+        response.setCredentialJson(credentialJson);
+        response.setProjectId(getAccountId(credentialJson));
         return response;
     }
 
+    private String getAccountId(String credentialJson) {
+        try {
+            JsonNode jsonNode = JsonUtil.readTree(new String(Base64.decodeBase64(credentialJson.getBytes())));
+            return jsonNode.get("project_id").asText();
+        } catch (Exception e) {
+            return null;
```

Comment:
A log would be nice to have here

Reply:
I just did not added because this is running for old credentials as well and I thought this cause unnecessary logs

Reply:
Yeah, but there is no log about the result of this logic which means in production you will have no clue what is happening. I would add at least one debug log with the result of the method, but also other exceptions could happen during json processing that also won't be reflected in the log.